src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -200,76 +192,100 @@ public boolean isCommentNodesRequired() {
+        String line = getLines()[lineNo - 1];
+
+        if (line.length() > ast.getLastChild().getColumnNo() + 1) {
+            line = line.substring(ast.getLastChild().getColumnNo() + 2);
+        }
 
-        for (Integer lineNo : lines) {
-            final String line = getLines()[lineNo - 1];
-            final String lineBefore;
-            final TextBlock comment;
-            if (cppComments.containsKey(lineNo)) {
-                comment = cppComments.get(lineNo);
-                lineBefore = line.substring(0, comment.getStartColNo());
-            }
-            else {
-                final List<TextBlock> commentList = cComments.get(lineNo);
-                comment = commentList.get(commentList.size() - 1);
-                lineBefore = line.substring(0, comment.getStartColNo());
+        line = FORMAT_LINE.matcher(line).replaceAll("");
 
-                // do not check comment which doesn't end line
-                if (comment.getText().length == 1
-                        && !CommonUtil.isBlank(line
-                            .substring(comment.getEndColNo() + 1))) {
-                    continue;
-                }
-            }
-            if (!format.matcher(lineBefore).find()
-                && !isLegalComment(comment)) {
-                illegalLines.put(lineNo, comment.getStartColNo());
-            }
+        final String lineBefore = getLines()[lineNo - 1].substring(0, ast.getColumnNo());
+
+        // do not check comment which doesn't end line
+        if ((ast.getLineNo() != ast.getLastChild().getLineNo() || CommonUtil.isBlank(line))
+                && !format.matcher(lineBefore).find()
+                && !isLegalBlockComment(ast, comment)) {
+            log(ast, MSG_KEY);
         }
     }
 
     /**
-     * Checks if given comment is legal (single-line and matches to the
-     * pattern).
+     * Checks if block comment is legal and matches to the pattern.
      *
+     * @param ast Detail ast element to be checked.
      * @param comment comment to check.
      * @return true if the comment if legal.
      */
-    private boolean isLegalComment(final TextBlock comment) {
+    private boolean isLegalBlockComment(DetailAST ast, String comment) {
         final boolean legal;
 
         // multi-line comment can not be legal
-        if (legalComment == null || comment.getStartLineNo() != comment.getEndLineNo()) {
+        if (legalComment == null
+                || !TokenUtil.areOnSameLine(ast.getFirstChild(), ast.getLastChild())) {
             legal = false;
         }
         else {
-            String commentText = comment.getText()[0];
-            // remove chars which start comment
-            commentText = commentText.substring(2);
-            // if this is a C-style comment we need to remove its end
-            if (commentText.endsWith("*/")) {
-                commentText = commentText.substring(0, commentText.length() - 2);
-            }
+            String commentText = comment;
```

Comment:
This is too verbose. Please change to simple
```java
String commentText = comment.trim();
```

Reply:
Done